### DISCUSSION TITLE
2D vertex model: adding noise contribution to forces

### DISCUSSION THREAD

**Discussion by mmarinriera** [Chaste/Chaste#161] [Q&A] [answered]
Dear Chaste community,

I developed, back in the day, a modified version of the 2D vertex model
following the examples and tutorials on the Chaste website. Recently, I
picked up the project again since we want to include random noise in the
force vectors to assess the robustness of the simulation results.

My question is, is there any functionality in the Cell-based models
suite to include noise in the vertex models?

If not, I am not sure which would be the best way to proceed.

I think the easiest way to implement noise would be to add a random
vector to the net force vector for each vertex during force
calculations. An alternative way would be to add a random perturbation
to the line tension parameter for each edge, but I think this would be
quite complicated to implement in practice.

I am thinking of implementing the noise contribution in the
AddForceContribution method of my modified force class, but maybe it
makes more sense to add it using a simulation modifier class?

I would really appreciate any tips or advice you may have. Thank you
very much.

Have a nice weekend!

Miquel

**Comment by fcooper8472** [comment, marked as answer]
Hi Muquel,

It sounds as though you're trying to do something very similar to a random force used in this paper:
https://doi.org/10.1371/journal.pcbi.1005387

Code for that can be found here:
https://github.com/Chaste/CellBasedComparison2017/blob/main/src/CellSorting/RandomMotionForce.cpp

in the class "RandomMotionForce". Note the coefficient in the random force should depend on the timestep (see equation 12 in the paper).

There are also more sophisticated ways of adding the force. In particular, one drawback of the above is that two points nearby in space might move too far with respect to each other. We will shortly be adding a method that generates spatially correlated noise, but that's not  ready just yet.

**Reply by AlexFletcher** [reply]
Hi Miquel,

Just to add to Fergus's comment, if you would like to impose a stochastic _displacement_ to each vertex rather than force at each time step, then simulation modifiers are how you could do this (bearing in mind that if such displacements are too large, you run the risk of self-intersecting edges etc). Happy to provide more details if needed.

Best wishes,
Alex

**Reply by mmarinriera** [reply]
Hi Fergus, Alex,
Yes, the implementation in "RandomMotionForce" is what I was looking for. I successfully managed to add a random contribution to the net force in the vertices. Thank you for the very helpful advice and tips.

Best,
Miquel
